sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/DirectoryAPITest.groovy

```diff
@@ -76,6 +78,26 @@ class DirectoryAPITest extends APISpec {
         validateBasicHeaders(createResponse.getHeaders())
     }
 
+    def "Create defaults with options"() {
+        setup:
+        dc = fsc.getDirectoryClient(generatePathName())
+        def options = new DataLakePathCreateOptions()
+        def metadata = new HashMap<String, String>()
+        metadata.put("foo", "bar")
+        options.setMetadata(metadata)
+            .setRequestConditions(new DataLakeRequestConditions())
+            .setPathHttpHeaders(new PathHttpHeaders())
+            .setAccessControlList(pathAccessControlEntries)
```

Comment:
1. should we assert that these has indeed get set ?
2. do we have test where options is just `new DataLakePathCreateOptions()`? i.e. check if implemnetation is null safe?

Reply:
We don't need to set things like `.setRequestConditions(new DataLakeRequestConditions())` and `.setPathHttpHeaders(new PathHttpHeaders())` in the tests they aren't being used.